src/app/api/v1/auth/verify/route.tsx

```diff
@@ -1,11 +1,68 @@
 import { NextResponse, type NextRequest } from "next/server";
+import { SiweMessage } from "siwe";
+import {
+  generateJwt,
+  getScopeForUser,
+  getExpiryForUser,
+} from "@/app/lib/auth/serverAuth";
 
+// This should create a user id record if there is not one already
+// This should not generate an API key
 export async function POST(request: Request) {
+  const body = await request.json();
+  const { message, signature, nonce } = body;
+
+  // N.B. the below assumes the address we're validating is an
+  // EOA. For EIP-1271, a provider will be necessary
+  const siweObject = new SiweMessage(message);
+  const verification = await siweObject.verify({
+    signature: signature,
+    nonce: nonce,
+  });
+
+  if (!verification.success) {
+    return NextResponse.json(
+      { message: `Invalid signature ${verification.error}` },
+      { status: 401 }
+    );
+  }
+
+  // create or update user record
+  let user = await prisma.api_user.findFirst({
+    where: {
+      address: verification.data.address,
+    },
+  });
+
+  if (!user) {
+    // TODO: extract into /common/users
+    user = await prisma.api_user.create({
+      data: {
+        address: verification.data.address,
+        // TODO: chain id is probably not important, and the abstraction
+        // might need to be rethought
+        chain: {
+          connect: {
+            id: "10",
+          },
+        },
+        enabled: true,
+        description: "Created by SIWE verification",
+      },
+    });
+  }
```

Comment:
is this just for record keeping? if so, we could do this db call async

Reply:
More or less; the idea is that we create a user ID for monitoring purposes. I can make this more concurrent, since it's not essential for authentication

Reply:
actually; we do need this in the auth flow to generate the user ID for the JWT request.